### PR TITLE
Port clients of ScrollableMixedWidgetList to LazyBlock

### PR DESCRIPTION
**Pull request by abarth**
LazyBlock is going to replace ScrollableMixedWidgetList at some point.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/lib/src/widgets/lazy_block.dart [hunk 1]
```diff
@@ -124,7 +136,7 @@ class _RenderLazyBlock extends RenderVirtualViewport<_LazyBlockParentData> {
       child.parentData = new _LazyBlockParentData();
   }
 
-  double _noIntrinsicExtent() {
+  bool _debugThrowIfNotCheckingIntrinsics() {
     assert(() {
       if (!RenderObject.debugCheckingIntrinsics) {
         throw new UnsupportedError(
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/lazy_block.dart]: This assertion text should probably be updated a little. :-)

Also, please remove the second \n while you're here.

File: packages/flutter/lib/src/widgets/lazy_block.dart [hunk 2]
```diff
@@ -259,13 +274,18 @@ class _LazyBlockElement extends RenderObjectElement {
...
-      for (int i = 0; i < widgets.length; ++i)
-        widgets[i] = builder(this, _firstChildLogicalIndex + i);
-      _children = updateChildren(_children, widgets);
+    LazyBlockDelegate newDelegate = newWidget.delegate;
+    LazyBlockDelegate oldDelegate = oldWidget.delegate;
+    if (newDelegate != oldDelegate && (newDelegate.runtimeType != oldDelegate.runtimeType || newDelegate.shouldRebuild(oldDelegate))) {
+      IndexedBuilder builder = newDelegate.buildItem;
+      List<Widget> widgets = <Widget>[];
+      for (int i = 0; i < widgets.length; ++i) {
+        Widget childWidget = builder(this, _firstChildLogicalIndex + i);
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/lazy_block.dart]: shouldn't you rewrap these in repaint boundaries?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/widgets/lazy_block.dart (hunk 2 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/lazy_block.dart]: Oops, yes.

**Pull request merged**